packages/modules/promotion/src/services/promotion-module.ts

```diff
@@ -475,6 +476,11 @@ export default class PromotionModuleService<
     const promotionsData: CreatePromotionDTO[] = []
     const applicationMethodsData: CreateApplicationMethodDTO[] = []
     const campaignsData: CreateCampaignDTO[] = []
+    const existingCampaigns = await this.campaignService_.list(
+      { id: data.map((d) => d.campaign_id).filter((id) => isString(id)) },
```

Comment:
thought: Could we do:
```suggestion
      { id: data.filter((d) => isPresent(d.campaign_id)) },
```